### PR TITLE
Fix crash when deleting layout items

### LINKED ISSUE #49573: QGIS crashes when deleting objects in print layout
### What is the bug or the crash?

Frequently QGIS have been crashing when deleting objects from print layout, happening most commonly with text boxes and images. I'm using the lastest version - 3.26.1 -  but this issue have been happening to me with several QGIS versions.

### Steps to reproduce the issue

1. Open a poreviously created print layout
2. Delete and object from it

### Versions

3.26.1 - Buenos Aires

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by elpaso**
Fix #49573

### CODE REVIEW

**Comment by elpaso** [comment]
CC @nyalldawson this fixes the crash but I couldn't really get to the root cause, do you have any idea where to look?

**Comment by nicogodet** [comment]
I confirm that it fixes the crash.

Any chance this could be backport ?

File: src/core/layout/qgslayoutitemundocommand.cpp [hunk 1]
```diff
@@ -55,10 +55,12 @@ void QgsLayoutItemUndoCommand::saveState( QDomDocument &stateDoc ) const
   QDomElement documentElement = stateDoc.createElement( QStringLiteral( "ItemState" ) );
 
   QgsLayoutItem *item = mLayout->itemByUuid( mItemUuid );
-  Q_ASSERT_X( item, "QgsLayoutItemUndoCommand::saveState", "could not retrieve item for saving state" );
-
-  item->writeXml( documentElement, stateDoc, QgsReadWriteContext() );
-  stateDoc.appendChild( documentElement );
+  if ( item )
+  {
+    QgsDebugMsg( QStringLiteral( "QgsLayoutItemUndoCommand::saveState: could not retrieve item %1 for saving state" ).arg( mItemUuid ) );
```
Comment (rouault) [inline src/core/layout/qgslayoutitemundocommand.cpp:60]: shouldn't this debug message be in an else clause, that is when item == nullptr ?

Comment (elpaso) [reply src/core/layout/qgslayoutitemundocommand.cpp]: O course, thank you for noticing!

**Comment by nicogodet** [comment]
Considering the potential data loss, can we skip the queued backport and target 3.22.12 ?

**Comment by elpaso** [comment]
> Considering the potential data loss, can we skip the queued backport and target 3.22.12 ?

I would say yes.

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
For reference (from a direct chat with @elpaso ):

This is caused by a design mistake i made -- I should have left all the undo stack related stuff out of the core layout classes
and only put these in the app code. Now there's way too much interplay between the configuration widgets and the undo stack
where one will affect the other in unpredictable ways. At some stage I'll need to redo this, but i guess in the meantime this fix can't make things worse! 😛

**Pull request merged**